packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -178,6 +183,49 @@ export class Dropdown extends BaseComponent<IDropdownProps, IDropdownState> {
     }
   }
 
+  protected _setMaxCalloutComponentHeight() {
+    let {calloutMaxHeight} = this.state;
+    if (this.props.pageSize && this.props.pageSize > 0) {
+      let totalItemCount = this._optionList.children.length;
+      let { selectedIndex } = this.state;
+
+      if (selectedIndex + this.props.pageSize > totalItemCount) {
+        selectedIndex = totalItemCount - this.props.pageSize;
+      }
+
+      if (selectedIndex < 0) {
+        selectedIndex = 0;
+      }
+
+      let maxHeight = null;
+      for (let i = selectedIndex; i < this.props.pageSize + selectedIndex; i++) {
+        let liItem = this.refs[Dropdown.Option + i] as HTMLLIElement;
+        if (liItem) {
+          maxHeight += liItem.offsetHeight;
+        }
+      }
+
+      if (maxHeight !== calloutMaxHeight) {
+        this.setState({ calloutMaxHeight: maxHeight });
+      }
+    }
+  }
+
+  protected _setCalloutWidth(hostElement?: HTMLDivElement) {
+    if (this.props.pageSize && this.props.pageSize > 0) {
+      let totalItemCount = this._optionList.children.length;
+      if (this.props.pageSize < totalItemCount) {
+        var mainPanel = hostElement.querySelector('.ms-Callout-main') as HTMLDivElement;
+        if (mainPanel) {
+          if (mainPanel.offsetWidth > this._dropDown.clientWidth) {
+            let offset = mainPanel.offsetWidth - this._dropDown.clientWidth + 2;
+            this.setState({ scrollbarOffset: offset });
```

Comment:
I don't think we really need scrollbarOffset, have you tried css `overflow` combinations to fix this ?